components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/util/MetadataConverter.java

```diff
@@ -34,8 +34,10 @@ FederatedAuthenticatorConfig getFederatedAuthenticatorConfig(Property properties
 
     boolean canHandle(FederatedAuthenticatorConfig federatedAuthenticatorConfig);
 
-    void saveMetadataString(int tenantId, String idpName, String metadata) throws IdentityProviderManagementException;
+    void saveMetadataString(int tenantId, String idpName, String fedAuthName, String metadata) throws IdentityProviderManagementException;
 
     void deleteMetadataString(int tenantId, String idPName)throws IdentityProviderManagementException;
 
-}
\ No newline at end of file
+    boolean canDelete(int tenantId, String idPName) throws IdentityProviderManagementException;
```

Comment:
let's add a default method, so that the API implementors will not be broken.

Reply:
Done.